### ISSUE TITLE
📢 Is It possible to install `MullvadVPN-2022.2_amd64.deb` file in `WSL2` ? 

### ISSUE THREAD

**Issue by wonkyDD** [mullvad/mullvadvpn-app#3795]
## Issue report
---
Operating system: `Window10` 10.0.19044 and  `Ubuntu 20.0.4` for wsl2

App version: 2022.2

---
#### TL;DR
> I want to use mullvad in cli on wsl2 same as window terminal

###### Error Log img below

![image](https://user-images.githubusercontent.com/81248624/181341052-8bd2794d-1267-4055-a1b6-754f9f429048.png)

As you can see from the above img,
`MullvadVPN-2022.2_amd64.deb` file is located in `/mnt/c/Users/User/Downloads`.

It says
> System has not been booted with systemd as init system (PID 1). Can't operate.

So I searched infos related to `systemd` and `init system` in wsl2 as keywords.
And I found question almost same with my question.

https://askubuntu.com/questions/1379425/system-has-not-been-booted-with-systemd-as-init-system-pid-1-cant-operate

Shortly, It says ..

>The core problem, as mention in the comments, is that <b><i>WSL doesn't use Systemd</i></b>, even in distributions where it is the default. Instead, WSL currently uses its own /init process as PID 1, which performs a few WSL-specific tasks . . .

I finally concluded that basically it isn't possible to use mullvad  in wsl2
because wsl2 doesn't use systemd as init system and error log says it needs systemd for boot system! Both confilcts each other 😱

For now, I'm confused and have no idea absolutely 😂
What am I supposed to do? What am I missing?

Thank you for reading.

### Edit: When I type `mullvad status` on wsl2 using ubuntu terminal 

![image](https://user-images.githubusercontent.com/81248624/181344691-08e7de6e-8927-4384-b495-9bd7de37734c.png)

**Comment by pinkisemils** [comment]
We don't have support for running the daemon as a standalone process, but if you were to just run it from it's install directory, it should *run*. I'm not sure how WSL networking is set up and whether any of the security guarantees our client tries to uphold will stand, but it might be possible to ensure that all network traffic from your WSL VM is sent through our tunnel.

**Comment by wonkyDD** [comment]
It's hard for me to figure out two sentences that you said..

> We don't have support for running the daemon as a standalone process

and
> if you were to just run it from it's install directory, it should run




# **<b>Former one</b>**
---

https://mullvad.net/en/help/install-mullvad-app-linux/
Following the above url explanation,
I typed `sudo /opt/Mullvad\ VPN/resources/mullvad-daemon -v` in ubuntu terminal on wsl2.

And It shows error like img below
![image](https://user-images.githubusercontent.com/81248624/181612042-26cc86e2-df93-4549-ad4b-621515c2cb30.png)

So former one sounds like for me that
>There is no way to avoid error 
when trying to start `mullvad-daemon` on wsl2

Do I understand correctly? If not, what am I supposed to do?

# **<b>Latter one</b>**
---

I tried hard to understand but still struggling on this thing.

What do you mean `just run it from it's install directory` ? 🤔
Is there a way to run mullvad from install directory on wsl2 or linux?


I tested installing mullvad on ubuntu using docker, 
but it didn't work showing similar error logs above case...😥

Thank you for reading

**Comment by pinkisemils** [comment]
> What do you mean just run it from it's install directory ? thinking
Is there a way to run mullvad from install directory on wsl2 or linux?

In the logs you show, it seems that the daemon is shut down due to receiving a `SIGTERM`. It also helps to launch it with the current working directory being it's install directory - like so `(cd /opt/Mullvad\ VPN/resources/; sudo ./mullvad-daemon -v)`. It will run in the foreground of course, but it should still work just fine. You should be able to control it via the CLI from a different shell. 

What I mean by us not supporting running our client without systemd is that we don't officially support such configurations. However, to get the GUI to work, it needs a running daemon instance and technically there are no runtime dependencies on systemd in the daemon, so if you can start the daemon process in an appropriate way, it may work. 
However, I personally have not run our client in such a configuration, nor have I used WSL1, so it's difficult for me to speculate as to what would happen - whether routing would be broken or if the firewall rules break something critical about WSL.

**Comment by wonkyDD** [comment]
Now I got it what to do thanks to your kind reply 🙇‍♂️

As you said, I'll find some ways to start mullvad daemon process on WSL2 and
look up some materials related to firewall rules break !

Thanks a lot!

**Comment by pinkisemils** [comment]
> What do you mean just run it from it's install directory ? thinking
Is there a way to run mullvad from install directory on wsl2 or linux?

In the logs you show, it seems that the daemon is shut down due to receiving a `SIGTERM`. It also helps to launch it with the current working directory being it's install directory - like so `(cd /opt/Mullvad\ VPN/resources/; sudo ./mullvad-daemon -v)`. It will run in the foreground of course, but it should still work just fine. You should be able to control it via the CLI from a different shell. 

What I mean by us not supporting running our client without systemd is that we don't officially support such configurations. However, to get the GUI to work, it needs a running daemon instance and technically there are no runtime dependencies on systemd in the daemon, so if you can start the daemon process in an appropriate way, it may work. 
However, I personally have not run our client in such a configuration, nor have I used WSL2, so it's difficult for me to speculate as to what would happen - whether routing would be broken or if the firewall rules break something critical about WSL or if the firewall works at all.

**Comment by vulpicastor** [comment]
Not re-opening the issue, but I just want to chime in that systemd is actually now supported on WSL 2; see Microsoft's [blog post](https://devblogs.microsoft.com/commandline/systemd-support-is-now-available-in-wsl/). In particular, the feature has been backported to Windows 10 as the Windows Store distribution of WSL is now [considered the default](https://devblogs.microsoft.com/commandline/the-windows-subsystem-for-linux-in-the-microsoft-store-is-now-generally-available-on-windows-10-and-11/) for both Win 10 and 11. While I have not tested this, it's now possible that the Mullvad Linux app will work out of the box in WSL 2 with systemd enabled.

**Comment by wonkyDD** [comment]
@vulpicastor Thanks a lot for notice! So glad what WSL2 has achieved. I would checkout it works 👀
